src/FxCop/Desktop.Analyzers/Core/Security/DoNotUseInsecureCryptographicAlgorithmsAnalyzer.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+using Desktop.Analyzers.Common;
+
+
+namespace Desktop.Analyzers
+{
+    public abstract class DoNotUseInsecureCryptographicAlgorithmsAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string DoNotUseMD5RuleId = "CA5350";
```

Comment:
I updated code according to the feedback except michaelcfanning's one. I don't know what the next step will look like. So am I now waiting for your decisions? @genlu @michaelcfanning